R/editable_fun.R

```diff
@@ -1,81 +1,97 @@
-deepstate_editable_fun<-function(package_path,function_name){
-  deepstate_fun_create(package_path,function_name,sep="generation")  
-  deepstate_fun_create(package_path,function_name,sep="checks")  
-  }
+deepstate_editable_fun <- function(package_path, function_name) {
+  gen <- deepstate_fun_create(package_path, function_name, sep="generation")
+  chk <- deepstate_fun_create(package_path, function_name, sep="checks")
+  c(gen, chk)
+}
 
-##' @title  Generation Testharness compilation
-##' @param package_path path to the testpackage
+##' @title Generation test harness compilation and execution
+##' @param package_path path to the tested package
 ##' @param function_name function name in the package
-##' @description This function compiles the generation testharness.
+##' @param continue_on_missing if set to FALSE, terminates the execution if 
+##' at leas one range is missing
+##' @param verbose used to deliver more in depth information
+##' @description This function compiles and runs the generation test harness.
+##' The generation test harness contains user defined ranges 
 ##' @export
-deepstate_compile_generate_fun <-function(package_path,function_name){
+deepstate_compile_generate_fun <- function(package_path, function_name,
+                      continue_on_missing=TRUE, verbose=getOption("verbose")){
   inst_path <- file.path(package_path, "inst")
   test_path <- file.path(inst_path,"testfiles")
-  filename  <- paste0(function_name,"_DeepState_TestHarness_generation.cpp")
-  no.specified.range <- list()
+  filename <- paste0(function_name,"_DeepState_TestHarness_generation.cpp")
   fun_path <- file.path(test_path,function_name)
   file_path <- file.path(test_path,function_name,filename)
   makefile.path <- file.path(test_path,function_name)
   if(file.exists(fun_path)){
     harness_lines <- readLines(file_path,warn=FALSE)
-    #harness_lines <- gsub("","",harness_lines,fixed=TRUE)
     range_elements <- nc::capture_all_str(harness_lines,"//",
                                           fun_call=".*","\\(")
-    for(fun_range in range_elements$fun_call[1:(length(range_elements$fun_call)/2)]){
-      range_check <- grep(paste0(fun_range,"();"),harness_lines,value = TRUE,fixed=TRUE)[1]
-      variable = nc::capture_all_str(range_check,val = ".*","="," RcppDeepState_",
-                                     type=".*","\\(")
-      #system(paste0("\"grep -e ",fun_range,"(); ",filename," | wc -l\""))
-      no.specified.range[[variable$val]] <- paste(variable$val,":",variable$type)
-      
-      message(sprintf("The following ranges are not specified you still want to continue?"))
-      print(no.specified.range)
-      response <- readline(prompt="Enter y/n to continue/exit:\n")
-      if(response == 'y'){
-        fun_generated <- deepstate_fuzz_fun(fun_path,time.limit.seconds=2,sep="generation")
-        print(fun_generated)
-        final_res <- deepstate_analyze_fun(fun_path,sep="generation")
-        print(final_res)
-      }
```

Comment:
@tdhock Currently, `deepstate_compile_generate_fun` and `deepstate_compile_checks_fun` wait for user input in the case of a non-defined range or missing assert conditions, respectively, and ask the user whether to continue or not. I decided it would be better to remove this prompt and provide a new `continue_on_missing` argument to these functions that, when set to `FALSE`, terminates the execution if ranges/asserts are missing.

Reply:
great.